CRM/Contact/Page/AJAX.php

```diff
@@ -815,7 +815,7 @@ public static function isOrQuery() {
    *
    * @param int $cid
    * @param int $oid
-   * @param "dupe-nondupe|nondupe-dupe" $oper
```

Comment:
`@param dupenondupe|nondupedupe $oper` is defining that the param should be either a `dupenondupe` or `nondupedupe` class- not quite what was expected. 

Currently the de-facto standards (https://docs.phpdoc.org/guide/references/phpdoc/types.html and https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md only allow for `@param string @oper` which the available strings documented as a comment.

If prefered, we could do `'dupenondupe'|'nondupedupe'` which whilst non-standard, is a format supported by at least one tool (https://psalm.dev/docs/annotating_code/typing_in_psalm/). There is also discussion of other syntaxes here, but unfortunately without much consensus https://github.com/phpDocumentor/phpDocumentor/issues/557.

So probably sticking with  `@param string @oper`  for now is best. 

(note this applies to the change in `CRM/Core/BAO/CustomField.php` too)

Reply:
@braders yep that definiely seems to be the easier approach. I had it in mind that there was a way to list available strings that is agreed - I think the definitive place for standards is psr5 now - https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md - but I haven't read it through